### PR TITLE
feat: Add export_id field on Taxonomy

### PR DESCRIPTION
**Pull request by ChrisChV**
**Description**

This PR adds a new `export_id` field on `Taxonomy` model.

**Support information**

- Github issue: https://github.com/openedx/modular-learning/issues/183
- Internal Ticket: [FAL-3620](https://tasks.opencraft.com/browse/FAL-3620)

**Testing instruction**

- Verify that all tests pass and that they cover the requirements.

Before run migration

- Start a server with `python manage.py runsever`.
- Create a superuser with `python manage.py createsuperuser`.
- Go to [Taxonomy list](http://127.0.0.1:8000/admin/oel_tagging/taxonomy/) on Django Admin.
- Create two Taxonomies with the same name.
- Run the migrations: `python manage.py migrate`.
- Verify that the migrations have been executen without errors.
- On Django admin create a new Taxonomy and verify that you need to add an `export_id`
- Verify that you can edit the `export_id` on Django admin.
- Verify the validations of the `export_id`

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @ChrisChV! Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

File: openedx_tagging/core/tagging/import_export/api.py
Comment (ChrisChV) [inline openedx_tagging/core/tagging/import_export/api.py:124]: This generated a warning on the lint

File: openedx_tagging/core/tagging/migrations/0015_taxonomy_export_id.py
Comment (ChrisChV) [reply openedx_tagging/core/tagging/migrations/0015_taxonomy_export_id.py:10]: Thanks :+1:

File: tests/openedx_tagging/core/tagging/test_views.py
Comment (ChrisChV) [reply tests/openedx_tagging/core/tagging/test_views.py:84]: Make sense. I will do this in the API

File: openedx_tagging/core/tagging/migrations/0015_taxonomy_export_id.py
Comment (ChrisChV) [reply openedx_tagging/core/tagging/migrations/0015_taxonomy_export_id.py]: Updated here: https://github.com/openedx/openedx-learning/pull/145/commits/b341e9b7f75cdde070400354399b77ac8f09b799

File: openedx_tagging/core/tagging/rest_api/v1/serializers.py
Comment (ChrisChV) [reply openedx_tagging/core/tagging/rest_api/v1/serializers.py]: Updated here 4b695682bfbbcfa263b697db87f767c7807d304d

File: tests/openedx_tagging/core/tagging/test_views.py
Comment (ChrisChV) [reply tests/openedx_tagging/core/tagging/test_views.py]: Updated here 4b695682bfbbcfa263b697db87f767c7807d304d

File: openedx_tagging/core/tagging/rest_api/v1/views.py
Comment (pomegranited) [inline openedx_tagging/core/tagging/rest_api/v1/views.py:61]: nit:

```suggestion
    TODO: We need to add a perform_update and call the api update function when is created.
```

File: openedx_tagging/core/tagging/rest_api/v1/views.py
Comment (pomegranited) [inline openedx_tagging/core/tagging/rest_api/v1/views.py:65]: I don't think we need to flag this risk here, because PUT requests are handled by `update_import`, which updates the tags.

We don't currently have a view for updating Taxonomy fields (but we can add one when needed).

File: tests/openedx_tagging/core/tagging/test_api.py
Comment (pomegranited) [inline tests/openedx_tagging/core/tagging/test_api.py:256]: nit: this is a good opportunity to change these `Taxonomy.object.create`s to `tagging_api.create_taxonomy`.

**Review by pomegranited — APPROVED** [review]
Couple of nits/questions, but everything else looks perfect. 👍 

@ChrisChV Let me know if you need me to do a second review?

- [x] I tested this in my devstack using the PR test instructions
- [x] I read through the code
- [x] <strike>I checked for accessibility issues</strike> N/A here
- [x] Includes documentation -- adds help_text for new field, docstrings where needed
- [x] User-facing strings are extracted for translation

File: openedx_tagging/core/tagging/rest_api/v1/views.py
Comment (ChrisChV) [reply openedx_tagging/core/tagging/rest_api/v1/views.py:65]: What I understand is that `update_import` is used with the url `/tagging/rest_api/v1/taxonomy/:pk/tags/import/plan` with PUT, but there is a view on `api/tagging/v1/taxonomy/:pk` with PUT to update the taxonomy.
I realized this because there are [tests for this view](https://github.com/openedx/openedx-learning/blob/2ec291845887b3db6bb56913ce1b12654d7e2574/tests/openedx_tagging/core/tagging/test_views.py#L413). I put this warning in case in the future the update of taxonomies is implemented in the UI

File: tests/openedx_tagging/core/tagging/test_api.py
Comment (ChrisChV) [reply tests/openedx_tagging/core/tagging/test_api.py:256]: Yep :+1:

Comment (ChrisChV) [reply tests/openedx_tagging/core/tagging/test_api.py]: Updated here: https://github.com/openedx/openedx-learning/pull/145/commits/f53025c0f6868443006abdc7a55ccd2a65149d1a

File: openedx_tagging/core/tagging/rest_api/v1/views.py
Comment (ChrisChV) [reply openedx_tagging/core/tagging/rest_api/v1/views.py]: Updated [here](https://github.com/openedx/openedx-learning/pull/145/commits/f53025c0f6868443006abdc7a55ccd2a65149d1a#diff-215d20c663dcd5727c14740a72d09fe6fd22e3a92fd08db46036e2e799699917R61)

**Review by bradenmacdonald — APPROVED** [review]
LGTM!

File: openedx_tagging/core/tagging/rest_api/v1/views.py
Comment (pomegranited) [reply openedx_tagging/core/tagging/rest_api/v1/views.py]: Ok ya, I didn't realise that. But cool, thank you for noticing and adding that warning.

**Comment by openedx-webhooks** [comment]
@ChrisChV 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/openedx-learning&entry.867055334=https://github.com/openedx/openedx-learning/pull/145&entry.1484655318=https://github.com/ChrisChV&entry.752974735=2024-01-26+16:07&entry.1917517419=2024-02-01+23:48&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**
